### DISCUSSION TITLE
QNAP Compatability

### DISCUSSION THREAD

**Discussion by mdomanic** [PartialVolume/shredos.x86_64#195] [General]
Has anyone had experience in using ShredOS for wiping drives within a QNAP NAS system?  I currently have a TS-877 i was hoping to wipe with drives installed.  ShredOS boots fine but does not see any of the attached drives.  There is incredibly limited BIOS settings that can be changed, and none that handle RAID/SATA/AHCI.

Anyone have any insight or possible other solutions for this system?  I specifically need to complete a DoD wipe.

Thanks!

**Comment by PartialVolume** [comment]
I don't have any experience of that hardware but on one QNAP manual I was reading (not necessarily for TS-877) it seemed to rely on some of their own software to do the configuration. The only option that seemed to make sense was putting the drives in 'single drive' mode.

**Comment by PartialVolume** [comment]
Can you also post the output of `lspci` which hopefully will identify the controller and tell me if we even have a driver for that.

**Comment by PartialVolume** [comment]
I just flicked through the manual for setting up one of these for the TS-877 [qts5.1.x-ug-en-us.pdf](https://github.com/PartialVolume/shredos.x86_64/files/13752547/qts5.1.x-ug-en-us.pdf)

To be honest I'd probably just take the discs out and wipe them on a normal SATA system. Maybe somebody that knows one of these NAS servers inside out may be able to tell you whether the disc controller provides disk pass through. Even so the output of `lspci` may prove useful to identify the controller and find out further information about it.

**Comment by mdomanic** [comment]
Here's the lspci output

00:00.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Complex
00:00.2 IOMMU: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) I/O Memory Management Unit
00:02.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Host Bridge
00:02.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Port
00:02.4 PCI bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Port
00:02.5 PCI bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Port
00:03.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Host Bridge
00:03.1 PCI bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Port
00:03.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Root Port
00:08.0 Encryption controller: Advanced Micro Devices, Inc. [AMD] Carrizo Platform Security Processor
00:09.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Carrizo Audio Dummy Host Bridge
00:09.2 Audio device: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Audio Controller
00:10.0 USB controller: Advanced Micro Devices, Inc. [AMD] FCH USB XHCI Controller (rev 20)
00:11.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 49)
00:12.0 USB controller: Advanced Micro Devices, Inc. [AMD] FCH USB EHCI Controller (rev 49)
00:14.0 SMBus: Advanced Micro Devices, Inc. [AMD] FCH SMBus Controller (rev 4a)
00:14.3 ISA bridge: Advanced Micro Devices, Inc. [AMD] FCH LPC Bridge (rev 11)
00:18.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 0
00:18.1 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 1
00:18.2 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 2
00:18.3 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 3
00:18.4 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 4
00:18.5 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h (Models 60h-6fh) Processor Function 5
01:00.0 SATA controller: Marvell Technology Group Ltd. 88SE9235 PCIe 2.0 x2 4-port SATA 6 Gb/s Controller (rev 11)
02:00.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
03:01.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
03:03.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
03:05.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
03:07.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
05:00.0 Ethernet controller: Intel Corporation I211 Gigabit Network Connection (rev 03)
07:00.0 USB controller: ASMedia Technology Inc. ASM1042A USB 3.0 Host Controller
08:00.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
09:01.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
09:03.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
09:05.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
09:07.0 PCI bridge: ASMedia Technology Inc. ASM1184e PCIe Switch Port
0a:00.0 Ethernet controller: Intel Corporation I211 Gigabit Network Connection (rev 03)
0c:00.0 Ethernet controller: Intel Corporation I211 Gigabit Network Connection (rev 03)
0d:00.0 Ethernet controller: Intel Corporation I211 Gigabit Network Connection (rev 03)
0e:00.0 SATA controller: Marvell Technology Group Ltd. 88SE9235 PCIe 2.0 x2 4-port SATA 6 Gb/s Controller (rev 11)
0f:00.0 VGA compatible controller: NVIDIA Corporation GK106 [GeForce GTX 645 OEM] (rev a1)
0f:00.1 Audio device: NVIDIA Corporation GK106 HDMI Audio Controller (rev a1)

**Comment by PartialVolume** [comment]
Thanks, just checked the marvel support for `01:00.0 SATA controller: Marvell Technology Group Ltd. 88SE9235 PCIe 2.0 x2 4-port SATA 6 Gb/s Controller (rev 11)` on your system.

We have Marvel SATA support installed however the information for the driver is as follows
```
CONFIG_SATA_MV:                                                                                                                    │  
  │                                                                                                                                    │  
  │ This option enables support for the Marvell Serial ATA family.                                                                     │  
  │ Currently supports 88SX[56]0[48][01] PCI(-X) chips,                                                                                │  
  │ as well as the newer [67]042 PCI-X/PCIe and SOC devices.
```
and also
```
CONFIG_PATA_MARVELL:                                                                                                               │  
  │                                                                                                                                    │  
  │ This option enables limited support for the Marvell 88SE61xx ATA                                                                   │  
  │ controllers. If you wish to use only the SATA ports then select                                                                    │  
  │ the AHCI driver alone. If you wish to the use the PATA port or                                                                     │  
  │ both SATA and PATA include this driver.
```
It doesn't explicitly state 88SE9235 so this might be why the discs don't show up. I can't be sure though.

**Comment by PartialVolume** [comment]
However, according to the [Linux for Hardware](https://linux-hardware.org/?id=pci:1b4b-9235-1b4b-9235) site 88SE9235 is supported by the ahci.drv (CONFIG_SATA_AHCI) driver, so in theory should work as that driver is also enabled in ShredOS.

```
CONFIG_SATA_AHCI:                                                                                                                  │  
  │                                                                                                                                    │  
  │ This option enables support for AHCI Serial ATA.
```
